components/engine_test/src/lib.rs

```diff
@@ -156,25 +265,28 @@ pub mod kv {
                 ));
             }
             let tablet_path = self.tablet_path(id, suffix);
-            let tablet_path = tablet_path.to_str().unwrap();
-            let kv_engine = KvTestEngine::new_kv_engine_opt(
-                tablet_path,
-                self.db_opt.clone(),
-                self.cf_opts.clone(),
-            )?;
+            let kv_engine = self.inner.create_tablet(&tablet_path)?;
             reg.insert((id, suffix), kv_engine.clone());
+
+            if let Some((old_suffix, _)) = reg_latest.get(&id) {
+                if *old_suffix < suffix {
+                    reg_latest.insert(id, (suffix, kv_engine.clone()));
+                }
+            } else {
+                reg_latest.insert(id, (suffix, kv_engine.clone()));
+            }
```

Comment:
You can use https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry to simplify it.

Reply:
Good suggestion.